### DISCUSSION TITLE
Hello! Help please! :)

### DISCUSSION THREAD

**Discussion by Alheimurine** [telegraf/telegraf#1899] [Q&A] [answered]
How to properly build a chain of questions and answers in telegraf.js?
Which method is better to use?
In my case, I'm stuck on answering the second question. More precisely answering the second question - the answer to the first is overwritten.. :(
Help me please.
```
bot.action('sellForm', async (ctx) => {
    try {
        await ctx.reply('Марка твоего авто?')
        bot.hears(/.*/, async (ctx) => {
            sellCarInfo.mark = ctx.update.message.text
            await ctx.reply('Модель твоего авто?')
        })
        bot.hears(/.*/, async (ctx) => {
            sellCarInfo.model = ctx.update.message.text
            await ctx.reply('Год выпуска твоего авто?')
        })
        bot.hears(/.*/, async (ctx) => {
            sellCarInfo.year = ctx.update.message.text
            await ctx.reply('Пробег твоего авто?')
        })
        bot.hears(/.*/, async (ctx) => {
            sellCarInfo.mile = ctx.update.message.text
            await ctx.reply(`Вы хотите продать автомобиль: ${sellCarInfo.mark} - ${sellCarInfo.model}. Год выпуска: ${sellCarInfo.year}, пробег: ${sellCarInfo.mile}`,
            Markup.inlineKeyboard(
                [Markup.button.callback('Подтвердить', 'sellYes')]
            ))
        })
    } catch(e) {
        console.error(e)
    }
})
```

**Comment by MKRhere** [comment, marked as answer]
So you cannot nest listeners inside each other. The appropriate tool to use is Wizard scenes. Your code is already very close and can be easily rewritten as a Wizard.

See this example: https://github.com/feathers-studio/telegraf-docs/blob/master/examples/wizards/wizard.ts

And this example for how you could filter updates: https://gist.github.com/MKRhere/183e6a9b95b6d2fe7f1e0ebad49e4045

**Reply by Alheimurine** [reply]
It is really easy solution. But i dont know much .ts. I had to figure it out a little and redo it myself. TY, friend! :)

**Reply by MKRhere** [reply]
I'm glad you were able to, but you can always paste a TS file here and get the JS version:

https://www.typescriptlang.org/play

TS is just JS + types
